kratos/geometries/brep_face_curve.h

```diff
@@ -0,0 +1,359 @@
+     */
+    template<class TOtherPointType, class TOtherPointEmbeddedType> explicit BrepFaceCurve(
+        BrepFaceCurve<TOtherPointType, TOtherPointEmbeddedType> const& rOther )
+        : BaseType( rOther )
+    {
+    }
+
+    /// Destructor
+    ~BrepFaceCurve() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    BrepFaceCurve& operator=( const BrepFaceCurve& rOther )
+    {
+        BaseType::operator=( rOther );
+        return *this;
+    }
+
+    /**
+     * Assignment operator for geometries with different point type.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    template<class TOtherPointType, class TOtherPointEmbeddedType>
+    BrepFaceCurve& operator=( BrepFaceCurve<TOtherPointType, TOtherPointEmbeddedType> const & rOther )
+    {
+        BaseType::operator=( rOther );
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename BaseType::Pointer Create( PointsArrayType const& ThisPoints ) const override
+    {
+        return typename BaseType::Pointer( new BrepFaceCurve( ThisPoints ) );
+    }
+
+    ///@}
+    ///@name Geometrical Operations
+    ///@{
+
+    /*
+    * @brief This method maps from dimension space to working space.
+    * @param rResult array_1d<double, 3> with the coordinates in working space
+    * @param LocalCoordinates The local coordinates in dimension space
+    * @return array_1d<double, 3> with the coordinates in working space
+    * @see PointLocalCoordinates
+    */
+    CoordinatesArrayType& GlobalCoordinates(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rLocalCoordinates
+    ) const override
+     {
+        CoordinatesArrayType surface_coordinates(3, 0.0);
+        mpNurbsCurve->GlobalCoordinates(surface_coordinates, rLocalCoordinates);
+        mpNurbsSurface->GlobalCoordinates(rResult, surface_coordinates);
+
+        return rResult;
+    }
+
+    /**
+    * Returns whether given arbitrary point is inside the Geometry and the respective
+    * local point for the given global point
+    * @param rPoint The point to be checked if is inside o note in global coordinates
+    * @param rResult The local coordinates of the point
+    * @param Tolerance The  tolerance that will be considered to check if the point is inside or not
+    * @return True if the point is inside, false otherwise
+    */
+    bool IsInside(
```

Comment:
I'm not sure if that's the behavior that people would expect when they call `IsInside` with a 3D-Point.

`std::numeric_limits<double>::epsilon()` is not a good value for a tolerance check. (@pooyan-dadvand this is done inside Kratos at many places)

Reply:
> `std::numeric_limits<double>::epsilon()` is not a good value for a tolerance check. (@pooyan-dadvand this is done inside Kratos at many places)

I know and you are right that it gives trouble on not near 1 values. Any suggestion? What you use?

Reply:
Actually, this function is only a hard simplification. I guess it will need more efforts, including the projections coming from #5477

Reply:
mentioning @loumaloumega who i think is a heavy user of this function

Reply:
@pooyan-dadvand specially for values around 1 this check does not make sense. For geometries I use the drawing tolerance given by the user. You can not give a general value for that since it depends on what your are drawing and which units you are using.

@tteschemacher at the moment it just checks if a given local parameter is inside the parameter domain of the curve(?). But it should check if the distance of a given global point to the geomertry is within a tolerance(?) In that case I would throw a not-yet-implemented exception until the projection works.

@RiccardoRossi Implementing this function as a member function of the geometry is not very effective if you plan to call the function many times. We have this situation in the CAD-Reconstruction where we are doing a lot of projections onto NURBS surfaces and curves. To avoid the overhead of recomputing temporary data we use classes for the projections.

Reply:
@oberbichler, are you suggesting to pass the epsilon in all methods or to have a member variable for that in geometry?

Reply:
It is a value that you specify when you create a CAD document so it would make sense to store it in the model. Store it for each geometry seems to be an overkill. What happens if two geometries in the same model have different tolerances?

You can not specify this tolerance in general but usually this is a value between 0.1 and 0.0001. `std::...::epsilon` is 2.22045e-16. That would kill every algorithm.

Reply:
…but for the moment I prefer to pass the argument to each function. At least I am doing this at the moment.

Reply:
Having the tolerance in the modelpart make sense to me.

Reply:
as i understand this conversation is resolved right?

in any case this is about the "IsInside" function of the geometry and not about the specific addition in this PR so unless there is no resistence over the next hours on my side i would approve

Reply:
It was also about the specific addition in this PR. IsInside is not computing the expected result. But maybe we just leave it as some dummy-code…

Reply:
I added an error, thus, we can discuss this separately.